src/types/happ.rs

```diff
@@ -0,0 +1,177 @@
+use anyhow::{Context, Result};
+use holochain_types::dna::AgentPubKey;
+use holochain_types::prelude::ActionHashB64;
+use holochain_types::prelude::AgentPubKeyB64;
+use holochain_types::prelude::MembraneProof;
+use holofuel_types::fuel::Fuel;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::HashSet;
+use std::fs::File;
+use std::time::Duration;
+use std::{collections::HashMap, env};
+use tracing::{trace, warn};
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct PublisherPricingPref {
+    pub cpu: Fuel,
+    pub storage: Fuel,
+    pub bandwidth: Fuel,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct DnaResource {
+    pub hash: String, // hash of the dna, not a stored dht address
+    pub src_url: String,
+    pub nick: String,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct HostSettings {
+    pub is_enabled: bool,
+    pub is_host_disabled: bool,
+    pub is_auto_disabled: bool,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct PresentedHappBundle {
+    pub id: ActionHashB64,
+    pub provider_pubkey: AgentPubKeyB64,
+    pub is_draft: bool,
+    pub is_paused: bool,
+    pub uid: Option<String>,
+    pub bundle_url: String,
+    pub name: String,
+    pub categories: Vec<String>,
+    pub jurisdictions: Vec<String>,
+    pub exclude_jurisdictions: bool,
+    pub special_installed_app_id: Option<String>,
+    pub host_settings: HostSettings,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct HappPreferences {
+    pub max_fuel_before_invoice: Fuel,
+    pub max_time_before_invoice: Duration,
+    pub price_compute: Fuel,
+    pub price_storage: Fuel,
+    pub price_bandwidth: Fuel,
+    pub invoice_due_in_days: u8,
+    pub jurisdiction_prefs: Option<ExclusivePreferences>,
+    pub categories_prefs: Option<ExclusivePreferences>,
+}
+impl HappPreferences {
+    /// Save preferences to a file under {SL_PREFS_PATH}
+    /// which allows hpos-api to read current values
+    pub fn save(self) -> Result<Self> {
+        if let Ok(path) = env::var("SL_PREFS_PATH") {
+            trace!("Writing default servicelogger prefs to {}", &path);
+            // create or overwrite to a file
+            let file = File::create(&path)?;
+            serde_yaml::to_writer(file, &self).context(format!(
+                "Failed writing service logger preferences to file {}",
+                path
+            ))?;
+        };
+        Ok(self)
+    }
```

Comment:
Is this still in use? If this is a clean up this should be removed

Reply:
It is in use upon install of the hosted happs, in order to save the hard coded  happ preferences to a file.  To confirm, you're saying we no longer need to create and reference this preference file, correct?

Reply:
After your implementation of getting the preferences from HHA for SL, I don't think we have been using this file.
The only place would be the hpos-api, and if that's the case that needs to be updated since that will no longer be accurate

Reply:
OK.  It does look like it's being used there.  I'll make a PR to amend that out too.